addons/mass_mailing_event/models/mailing_mailing.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, api, _
+from odoo.exceptions import UserError
+
+
+class MassMailingCampaign(models.Model):
+    _inherit = "mailing.mailing"
+
+    @api.constrains('mailing_model_id', 'mailing_domain')
+    def _check_domain_for_event_registration(self):
+        for mass_mailing in self:
+            if mass_mailing.mailing_model_name == 'event.registration' and 'event_id' not in mass_mailing.mailing_domain:
+                raise UserError(_('You must filter on the event to contact attendees'))
+
+    def _get_default_mailing_domain(self):
+        mailing_domain = super(MassMailingCampaign, self)._get_default_mailing_domain()
+        if self.mailing_model_name == 'event.registration' and not mailing_domain:
+            mailing_domain = self.env.context.get('default_mailing_domain', [])
```

Comment:
What is the purpose of this code ? Mailing domain is a computed editable field. It should be possible to support default values. Is there any compute / onchange thingy ? In which case we should find a generic solution.

Reply:
@tde-banana-odoo 
The method `_get_default_mailing_domain` is called to set the initial value of the domain.
If I do not overwrite it, it will not take the default values in the context, but the value returned in the parent method `_get_default_mailing_domain` :/

Reply:
Well I suppose this is due to default get being called before computed fields are trigerred on form view.

Maybe we could simply support default_mailing_domain in the computed field.